src/jdk.jdeps/share/classes/com/sun/tools/jnativescan/Main.java

```diff
@@ -0,0 +1,203 @@
+        this.err = err;
+    }
+
+    private void printError(String message) {
+        err.println("ERROR: " + message);
+    }
+
+    private void printUsage()  {
+        out.print("""
+            Use 'jnativescan --help' for help
+            """);
+    }
+
+    private void printVersion() {
+        out.println("jnativescan " + Runtime.version());
+    }
+
+    public int run(String[] args) {
+        if (args.length < 1) {
+            printUsage();
+            return FATAL_ERROR_CODE;
+        }
+
+        try {
+            String[] expandedArgs = expandArgFiles(args);
+            parseOptionsAndRun(expandedArgs);
+        } catch (JNativeScanFatalError fatalError) {
+            printError(fatalError.getMessage());
+            if (DEBUG) {
+                fatalError.printStackTrace(err);
+            }
+            return FATAL_ERROR_CODE;
+        } catch (Throwable e) {
+            printError("Unexpected exception encountered");
+            e.printStackTrace(err);
+            return FATAL_ERROR_CODE;
+        }
+
+        return SUCCESS_CODE;
+    }
+
+    private void parseOptionsAndRun(String[] expandedArgs) throws JNativeScanFatalError {
+        OptionParser parser = new OptionParser(false);
+        OptionSpec<Void> helpOpt = parser.acceptsAll(List.of("?", "h", "help"), "help").forHelp();
+        OptionSpec<Void> versionOpt = parser.accepts("version", "Print version information and exit");
+        OptionSpec<String> classPathOpt = parser.accepts(
+                "class-path",
+                "The class path as used at runtime")
+                .withRequiredArg();
+        OptionSpec<String> modulePathOpt = parser.accepts(
+                "module-path",
+                "The module path as used at runtime")
+                .withRequiredArg();
+        OptionSpec<String> releaseOpt = parser.accepts(
+                "release",
+                "The runtime version that will run the application")
+                .withRequiredArg();
+        OptionSpec<String> addModulesOpt = parser.accepts(
+                "add-modules",
+                "List of root modules to scan")
+                .withRequiredArg();
+        OptionSpec<Void> printNativeAccessOpt = parser.accepts(
+                "print-native-access",
+                "print a comma separated list of modules that may perform native access operations." +
+                        "ALL-UNNAMED is used to indicate unnamed modules.");
+
+        OptionSet optionSet;
+        try {
+            optionSet = parser.parse(expandedArgs);
+        } catch (OptionException oe) {
+            throw new JNativeScanFatalError("Parsing options failed: " + oe.getMessage(), oe);
+        }
+
+        if (optionSet.has(helpOpt)) {
+            out.println("""
+                The jnativescan tool can be used to find methods that may access native functionality when
+                run. This includes methods that call restricted methods, and 'native' method declarations.
```

Comment:
I think it would be more readable if we said "This includes restricted method calls and `native` method declarations`. The way it is now it seems (for some weird subjective reason) that the sentence is saying "methods ... that call ... native method declarations" :-)

Reply:
Oh, the oxford comma has failed me :D